aws/snapshot.go

```diff
@@ -29,14 +37,30 @@ func getAllSnapshots(session *session.Session, region string, excludeAfter time.
 
 	var snapshotIds []*string
 	for _, snapshot := range output.Snapshots {
-		if excludeAfter.After(*snapshot.StartTime) {
+		if excludeAfter.After(*snapshot.StartTime) && !SnapshotHasAWSBackupTag(snapshot.Tags) {
 			snapshotIds = append(snapshotIds, snapshot.SnapshotId)
 		}
 	}
 
 	return snapshotIds, nil
 }
 
+// Check if the image has an AWS Backup tag
+// Resources created by AWS Backup are listed as owned by self, but are actually
+// AWS managed resources and cannot be deleted here.
+func SnapshotHasAWSBackupTag(tags []*ec2.Tag) bool {
```

Comment:
Nice - did we want to merge this first to keep the changes flowing and then figure out the go-commons placement later? Or vice versa?

Reply:
I think let's roll forward as is for now. The number of resources this touches is fairly small so we can do it on a 1 by 1 basis for now, I think.